### DISCUSSION TITLE
I am using head objects those are interactive with context in my project. I was able to provide this using next/head when using pages router but i can't do this when using app router.

### DISCUSSION THREAD

**Discussion by veyseltoprak** [vercel/next.js#50834] [Ideas]
### Goals

1. My main goal is to be able to use something like next/head when developing an app with app router. 
2. 
3.

### Non-Goals

1.
2.
3.

### Background

I am using head objects those are interactive with context in my project. I was able to provide this using next/head when using pages router but i can't do this when using app router.  Metadata is not alternative for this. 

### Proposal

Actually I want next/head to work in the same way.

**Comment by timothyerwin** [comment]
yeah, I think this is a blocker for our migration. we need head access for a number of reasons, not just metadata. for example we add custom scripts or stylesheets and we need to use data we fetched for the route as well in the meta tags and don't want to fetch it twice.

it feels like a very poor design decision.

**Comment by ravdz** [comment]
Same here, I would like to use classic next/head component and just pass there some props instead of fetching once again. I hope it will change somehow in future versions.
